### ISSUE TITLE
Add directory clone detection

### ISSUE THREAD

**Issue by nicolasboulay** [pkolaczk/fclones#51]
Sometime i keep 2 times the content of a CF card full of video or photos. 

It will be great to have a detection of directory which all content are already present somewhere else.

**Comment by felciano** [comment]
Agreed -- this would be a massive convenience for deduplication efforts.

**Comment by ashaikh** [comment]
This would be great to have and a feature to merge very similar directories. Your tool is fantastic, fastest I have ever seen and I have tried most of them on mac, windows, linux. 

For features you may want to look at Duplicate File Finder for Mac, that has the most usable set of features to combine and eliminate duplicates while doing a system cleanup. It is much slower than yours but it lets you clean things up very intuitively.

**Comment by shakfu** [comment]
Totally agree. I also had duplicated folders which i had to figure on a file by file basis.

I guess after all files checks are completed, it would involve sorting by path and then a recursive comparison of hashes to determine which directories are duplicate.

**Comment by JohniFi** [comment]
_First of all, great software, thanks!_

This would indeed be very helpful with big collections (I just did a scan for a friend of mine: "Total: 5288645275774 B (5.3 TB) in 919622 files in 238361 groups"). In this case I won't just let fclones automatically delete thousands of files but would rather like to find the biggest "whole" folders that match and delete them manually in the file explorer after manual revision to be sure to not destroy anything (e.g. delete files from video/music software project folders, that would corrupt the project).

**Idea:**
I guess for this kind of tool you have to scan for duplicates and uniques, maybe store the folder structure in a BTree and then find folders that have redundant content without unique content and then traversing upwards in the BTree as much as possible to find the biggest folders that match.

**Comment by fugalh** [comment]
Related, and maybe an easier change, it would be nice to print out a pairwise list of directories that contain duplicates. Then I can work through that list and manually take care of folders of duplicates or mostly duplicates a folder at a time and potentially use fclones more intelligently on those two directories (e.g. grab just the unique or link/remove as appropriate for those two directories)

**Comment by TheRealGramdalf** [comment]
What would be required for this to be implemented? I'm staring down 21+TB of data which includes a lot of duplicate directories (multiple backup drives copied into a ZFS pool, containing things like duplicate CF/SD card backups, full windows install trees etc) and a duplicate folder feature would be massively helpful. I could possibly get by manually comparing folders which are likely to be duplicates, but a dedicated folder duplicate reporter would be ideal.

@pkolaczk Would this require a big architectural change, or is it relatively simple to implement? If it would require more in depth changes, would there be something simpler such as a list of directories that only contain duplicates or what @fugalh suggested?
